colossalai/zero/utils/zero_hook.py

```diff
@@ -92,18 +68,32 @@ def pre_bwd_exec(self, module: torch.nn.Module, input, output):
             for param in module.parameters(recurse=False):
                 colo_model_data_tensor_move_inline(param.colo_attr.sharded_data_tensor, self.computing_device)
 
-        # gather sharded parameters
-        if module.param_is_sharded:
-            tensor_list = []
-            for param in module.parameters(recurse=False):
-                assert hasattr(param, 'colo_attr')
-                tensor_list.append(param.colo_attr.sharded_data_tensor)
-            self.shard_strategy.gather(tensor_list, self.process_group)
-
-        # record memory statistics
+        # record model data statistics
         if self._memstarts_collector:
-            self._memstarts_collector.sample_memstats()
+            self._memstarts_collector.sample_model_data()
```

Comment:
why sample model data here?

Reply:
I think model data sampling should be done at the same time of overall data sampling.